clang/lib/AST/Interp/ByteCodeEmitter.cpp

```diff
@@ -228,6 +229,42 @@ void emit(Program &P, std::vector<std::byte> &Code, const Floating &Val,
   Val.serialize(Code.data() + ValPos);
 }
 
+template <>
+void emit(Program &P, std::vector<std::byte> &Code,
+          const IntegralAP<false> &Val, bool &Success) {
+  size_t Size = Val.bytesToSerialize();
+
+  if (Code.size() + Size > std::numeric_limits<unsigned>::max()) {
```

Comment:
Should this be looking at `numeric_limits<signed>` instead?

Reply:
Why?

Reply:
Because I was confused, that's why! (I was mistakenly thinking this was validating the value fit within range, but noooooo that's not what it's doing at all.)

Reply:
Ah, ok. No this checks if the additional value would overflow `unsigned`. :)